torch/_decomp/decompositions.py

```diff
@@ -2058,13 +2053,11 @@ def _compute_upsample_nearest_indices(input, output_size):
         # scale = isize / osize
         # input_index = floor(output_index * scale)
         # Same as OpenCV INTER_NEAREST
-        osize = sym_float(output_size[d])
-        output_indices = torch.arange(
-            sym_int(osize), dtype=input.dtype, device=input.device
-        )
-        isize = sym_float(input.shape[-num_spatial_dims + d])
-        scale = isize / osize
-        input_indices = torch.floor(output_indices * scale).to(torch.int64)
+        osize = output_size[d]
+        output_indices = torch.arange(osize, dtype=input.dtype, device=input.device)
+        isize = input.shape[-num_spatial_dims + d]
+        scale = 1 / scales[d] if scales[d] is not None else isize / osize
+        input_indices = (output_indices * scale).to(torch.int64)
```

Comment:
This logic here makes me a little nervous, because it does feel like `1 / scales[d]` could end up being numerically different than when we recompute `isize / osize` here. But, it seems like maybe this is better? So the answer might be to update eager mode behavior too.

Reply:
If I had to guess, this is what's causing tests to fail. And we can't even fix it because we are matching OpenCV behavior exactly.

Reply:
yup, OpenCV compat lol

```
// see NOTE [ Nearest neighbor upsampling kernel implementation ]
__device__ __forceinline__ static int nearest_neighbor_compute_source_index(
    const float scale,
    int dst_index,
    int input_size) {
  // index_f32 = (output_index) * scale
  // input_index = round(index_f32)
  // Same as a buggy OpenCV INTER_NEAREST
  // We keep this method for BC and consider as deprecated.
  // See nearest_neighbor_exact_compute_source_index as replacement
  const int src_index =
      min(static_cast<int>(floorf((dst_index) * scale)), input_size - 1);
  return src_index;
}
```

But Triton supports floorf so it should be fine right?

Reply:
Ugh ok I'll probably need to do something different for upsampling nearest, that's fine